server/services/caldav/lib/calendar/calendar.syncAllUsers.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * @description Sync all users calendars.
+ * @returns {Promise} All calendars are sync.
+ * @example
+ * syncAllUsers();
+ */
+async function syncAllUsers() {
+  try {
+    const users = await this.gladys.user.get();
+    const service = await this.gladys.service.getLocalServiceByName('caldav');
+
+    return Promise.all(
```

Comment:
Be careful here, if you sync all users in parallel, you can spam the Caldav server and the Gladys instance can be rate-limited and lose access to the server for a short period of time.

Imagine we have 5 users in the family, having each 5-6 calendars, it's easily 30 requests/seconds to the provider, I don't think it's something we want.

There are several ways of rate limiting calls: 

- You can use bluebird [Promise.map](http://bluebirdjs.com/docs/api/promise.map.html) with a `concurrency` option.
- You can rate limit at the request level by creating a pool of request in the `request` module (see the maxSockets option https://github.com/request/request#requestoptions-callback)

Reply:
Indeed not the best way, I'll try to improve that.